oak_functions/client/java/src/AeadEncryptor.java

```diff
@@ -1,39 +1,92 @@
 package com.google.oak.functions.client;
 
-import com.google.crypto.tink.subtle.AesGcmJce;
+import com.google.oak.functions.client.Util;
+import java.lang.IllegalArgumentException;
 import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
 
 /**
  * Implementation of Authenticated Encryption with Associated Data (AEAD).
  * https://datatracker.ietf.org/doc/html/rfc5116
  */
 public class AeadEncryptor {
-    private AesGcmJce encryptor;
+    private static String ENCRYPTION_ALGORITHM = "AES";
+    private static String AEAD_ALGORITHM = "AES/GCM/NoPadding";
+    private static int KEY_LENGTH_BITS = 256;
+    private static int TAG_LENGTH_BITS = 128;
+    private static int INITIALIZATION_VECTOR_LENGTH_BYTES = 12;
 
-    public AeadEncryptor(byte[] key) throws GeneralSecurityException {
-        // TODO(#2120): Use a wrapper AEAD interface and don't use `tink.subtle`.
-        this.encryptor = new AesGcmJce(key);
+    private SecretKey key;
+    private SecretKeySpec keySpecification;
+
+    public AeadEncryptor(byte[] key) throws IllegalArgumentException {
+        if (key.length * 8 != KEY_LENGTH_BITS) {
+            throw new IllegalArgumentException(
+                String.format("Incorrect key length: %d, expected %d", key.length * 8, KEY_LENGTH_BITS)
+            );
+        }
+        this.key = new SecretKeySpec(key, 0, key.length, AEAD_ALGORITHM);
+        this.keySpecification = new SecretKeySpec(this.key.getEncoded(), ENCRYPTION_ALGORITHM);
     }
 
     /**
-     * Encrypts `data` using `AeadEncryptor::encryptor`.
-     * The resulting encrypted data is prefixed with a random 12 bit nonce.
+     * Encrypts `data` using AES-GCM.
+     * The resulting encrypted data is prefixed with a random 12 bit initialization vector.
      */
     public byte[] encrypt(byte[] data) throws GeneralSecurityException {
+        Cipher encryptor = Cipher.getInstance(AEAD_ALGORITHM);
+
+        // Create a random initialization vector.
+        byte[] initializationVector = generateInitializationVector(INITIALIZATION_VECTOR_LENGTH_BYTES);
+        GCMParameterSpec gcmParameterSpecification = new GCMParameterSpec(TAG_LENGTH_BITS, initializationVector);
+
+        // Initialize encryptor.
+        encryptor.init(Cipher.ENCRYPT_MODE, this.keySpecification, gcmParameterSpecification);
```

Comment:
nit: in java it is customary to not use `this.` for field access unless it is needed to differentiate from a conflicting local variable or argument. Same applies throughout these changes.

Reply:
Done